### DISCUSSION TITLE
Reaction Force on boundary

### DISCUSSION THREAD

**Discussion by manikumd22** [idaholab/moose#27325] [Q&A Modules: Solid mechanics]
Hi @GiudGiud 

I am trying to calculate the reaction force in the x-direction on the left face (fixed), but getting an error

**** Error***

[31m
*** ERROR ***
Each variable must be referenced by at least one active Kernel.
The following variable(s) lack an active kernel: resid_x [39m

application called MPI_Abort(MPI_COMM_WORLD, 1) - process 2

[31m
*** ERROR ***
Each variable must be referenced by at least one active Kernel.
The following variable(s) lack an active kernel: resid_x [39m

application called MPI_Abort(MPI_COMM_WORLD, 1) - process 3

Input File:

[GlobalParams]
  displacements = 'disp_x disp_y'
[]

[Mesh]
  [./base]
  type = FileMeshGenerator
  file = manik.msh
  [../]
[]

[Postprocessors]
  [./resid_x]
    type = SideAverageValue
    variable = resid_x
    boundary = 'left'
  [../]
[]

[Variables]
  [disp_x]
  []
  [disp_y]
  []
  [./resid_x]
    order = FIRST
    family = LAGRANGE
  [../]
[]

[Kernels]
  [sdx]
    type = StressDivergenceTensors
    variable = disp_x
    component = 0
  []
  [sdy]
    type = StressDivergenceTensors
    variable = disp_y
    component = 1
  []
[]

[BCs]
  [./left_x]
    type = DirichletBC
    variable = disp_x
    boundary = 'left'
    value = 0.0
  [../]
  [./left_y]
    type = DirichletBC
    variable = disp_y
    boundary = 'left'
    value = 0.0
  [../]
  [./right]
    type = DirichletBC
    variable = disp_x
    boundary = 'right'
    value = 1
  [../]
[]

[Materials]
  [elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    poissons_ratio = 0.3
    youngs_modulus = 1e6
  []
  [stress]
    type = ComputeFiniteStrainElasticStress
  []
[]

[Preconditioning]
  [smp]
    type = SMP
    full = true
  []
[]

[Executioner]
  type = Steady

  solve_type = 'newton'

  line_search = 'none'

  petsc_options = '-pc_svd_monitor'
  petsc_options_iname = '-pc_type'
  petsc_options_value = 'svd'

  nl_max_its = 500
  nl_abs_tol = 1e-1
  nl_rel_tol = 1e-1
  l_tol = 1e-1
[]

[Outputs]
  print_linear_residuals = false
  interval = 1
  csv = True
  perf_graph = false
  exodus = true
  file_base = ./inputfiles/manik_neohookean_reaction_force
[]

**Comment by GiudGiud** [comment]
Hello

what is this resid_x variable? what do you need that for?

Guillaume
